test/typescript/enhancers.ts

```diff
@@ -13,12 +13,17 @@ function dispatchExtension() {
 
   const enhancer: StoreEnhancer<{
     dispatch: PromiseDispatch
-  }> = createStore => <S, A extends Action = AnyAction>(
-    reducer: Reducer<S, A>,
-    preloadedState?: any
-  ) => {
+  }> = createStore => (reducer, preloadedState) => {
     const store = createStore(reducer, preloadedState)
-    return {
+
+    function replaceReducer<NewState, NewActions extends Action>(
+      nextReducer: Reducer<NewState, NewActions>
+    ) {
+      store.replaceReducer(nextReducer)
+      return newStore
+    }
```

Comment:
Many of the enhancer tests had incorrect `replaceReducer` methods. Previously they were actually returning the old store without the enhancer extensions applied instead of returning the new store with the enhancers applied. The fix in the types caught this bug.

Reply:
Also note that the boilerplate that is now required for any enhancer is significantly larger since `replaceReducer` has to return the store. If https://github.com/reduxjs/redux/pull/3772 gets merged, this boilerplate would reduce significantly.